### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by DenysGonchar**
swagger documentation can be found [here](https://esl.github.io/amoc_rest/?v=initial_implementation)

### CODE REVIEW

**Review by goddammit — APPROVED** [review]

File: rebar.config [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{deps, [
+    {cowboy, {git, "https://github.com/ninenines/cowboy.git", {tag, "2.7.0"}}},
+    {rfc3339, {git, "https://github.com/talentdeficit/rfc3339.git", {tag, "master"}}},
+    {jsx, {git, "https://github.com/talentdeficit/jsx.git", {tag, "2.9.0"}}},
+    {jesse, {git, "https://github.com/for-GET/jesse.git", {tag, "1.5.2"}}}
```
Comment (michalwski) [inline rebar.config]: Are there hex packages for the deps? Usually it's a bit faster and easier to cache locally a hex package. This is not a big deal but reduces built time.

File: src/amoc_rest_utils.erl [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+-spec to_qs(iodata() | atom() | number()) -> binary().
+
+to_qs(Name) ->
+    to_binary(Name).
+
+-spec to_binding(iodata() | atom() | number()) -> atom().
+
+to_binding(Name) ->
+    Prepared = to_binary(Name),
+    binary_to_atom(Prepared, utf8).
```
Comment (michalwski) [inline src/amoc_rest_utils.erl:80]: Do we have control over this? It's dangerous to cast binary to any atom.

**Review by michalwski — COMMENTED** [review]
Thanks for moving the generated REST code to this repo. I'm a bit afraid of the `binary_to_atom` function.

File: src/amoc_rest_utils.erl (hunk 2 above)
Comment (DenysGonchar) [reply src/amoc_rest_utils.erl:80]: This is generated code, however AFAIK it's not used anywhere. I tried to keep the changes to the generated code as low as possible

File: rebar.config (hunk 1 above)
Comment (DenysGonchar) [reply rebar.config]: rfc3339 will stay as is because it's not updated on hex

**Review by michalwski — APPROVED** [review]

**Pull request merged**